src/GraphBLAS-sharp.Backend/Vector/Vector.fs

```diff
@@ -0,0 +1,153 @@
+namespace GraphBLAS.FSharp.Backend
+
+open Brahma.FSharp
+open GraphBLAS.FSharp.Backend
+open Microsoft.FSharp.Control
+open Microsoft.FSharp.Quotations
+open GraphBLAS.FSharp.Backend.Common
+
+module Vector =
+    let zeroCreate (clContext: ClContext) (workGroupSize: int) =
+
+        let zeroCreate =
+            ClArray.zeroCreate clContext workGroupSize
+
+        fun (processor: MailboxProcessor<_>) (size: int) (format: VectorFormat) ->
+            match format with
+            | Sparse ->
+                let vector =
+                    { Context = clContext
+                      Indices = clContext.CreateClArray [| 0 |]
+                      Values = clContext.CreateClArray [| Unchecked.defaultof<'a> |]
+                      Size = size }
+
+                ClVectorSparse vector
+            | Dense -> ClVectorDense <| zeroCreate processor size
+
+    let ofList (clContext: ClContext) =
+        fun (format: VectorFormat) (elements: (int * 'a) list) ->
+            let indices, values =
+                elements
+                |> Array.ofList
+                |> Array.sortBy fst
+                |> Array.unzip
+
+            let resultLenght = (Array.max indices) + 1
```

Comment:
I think for dense vector this is incorrect. In dense case `ofList` operation should create vector with zeroes and assign non-zero values by given indices. This means that number of indices can be lesser than size of the array and `ofList` should take vector size as a parameter. Also, `indices` are sorted at this point so it is better to take last element of array than call `Array.max`